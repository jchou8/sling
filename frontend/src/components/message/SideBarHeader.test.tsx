import * as React from 'react'
import { createShallow, createMount } from '@material-ui/core/test-utils'
import SideBarHeader from './SideBarHeader'
import { User } from '../../types'

const logOutFn = jest.fn()

describe('<SideBarHeader />', () => {
  let shallow: Function
  let mount: Function
  beforeAll(() => {
    shallow = createShallow()
    mount = createMount()
  })

  it('should render properly', () => {
    const component = shallow(<SideBarHeader curUser={null} logOut={logOutFn} />)
    expect(component).toMatchSnapshot()
  })

  it('should display the user\'s name', () => {
    const user: User = {
      username: 'Jeff',
      id: 1,
      jwtToken: null
    }

    const component = shallow(<SideBarHeader curUser={user} logOut={logOutFn} />)
    expect(component.containsMatchingElement(<label>Jeff's sling</label>)).toBe(true)
  })

  it('should log the user out upon pressing the log out button', () => {
    const component = mount(<SideBarHeader curUser={null} logOut={logOutFn} />)
    component.find('button').simulate('click')
    expect(logOutFn).toHaveBeenCalled()
  })
})