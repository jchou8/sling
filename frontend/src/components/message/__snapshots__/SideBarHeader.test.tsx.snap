// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBarHeader /> should render properly 1`] = `ShallowWrapper {}`;
